### PR TITLE
ccv3.Client : replace usages of newHTTPRequet when possible

### PR DESCRIPTION
**Pull request by gmllt**
## Description of the Change

This PR replaces usages of ccv3.Client `newHTTPRequest` method in favor of ccv3.Client `MakeRequest`, `MakeListRequest` and `MakeRequestUploadAsync` methods for the following ccv3.Client methods : 
- CreateBuildpack
- DeleteBuildpack
- GetBuildpacks
- UpdateBuildpack
- UploadBuildpack
- uploadBuildpackAsynchronously
- GetApplicationEnvironment
- GetFeatureFlag
- GetFeatureFlags
- UpdateFeatureFlag
- GetApplicationManifest
- EntitleIsolationSegmentToOrganizations
- ShareServiceInstanceToSpaces
- ResourceMatch
- CreateApplicationTask
- GetApplicationTasks
- UpdateTaskCancel

## Why Is This PR Valuable?

This PR standardize interactions with the ccv3.Client, using `MakeRequest`, `MakeListRequest` and `MakeRequestUploadAsync` for the methods mentionned above as they are already used in others ccv3.Client methods.

## How Urgent Is The Change?

Those changes are not urgent but will help to get rid of a failure encountered during the entitlement of an IsolationSegment to an Organization using the cloudfoundry terraform provider ([cloudfoundry-community/terraform-provider-cloudfoundry#449](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/issues/449)).

### CODE REVIEW

**Review by sleungcy — APPROVED** [review]

**Comment by Thanhphan1147** [comment]
Hello, the PR looks ok to me. However I have a slight doubt since this means that we are diverging further from the client implementation at the upstream (cloudfoundry/cli). I think simply copying the implementations from the v7 or v8 branch would be a better solution. What do you think?

**Comment by gmllt** [comment]
@Thanhphan1147 I have updated the branch. I have copied implementations from the upstream, and extracted ressources as well.
Is this what you were thinking off ?

**Comment by Thanhphan1147** [comment]
Hello, looks good to me. Thanks

**Comment by gmllt** [comment]
@sleungcy, can you take a look at this PR again (i modified it to include @Thanhphan1147's recommendations) ?

**Review by Thanhphan1147 — APPROVED** [review]

**Comment by gmllt** [comment]
@sleungcy, can you take a look at this PR again (i modified it to include @Thanhphan1147's recommandations) ?

**Review by sleungcy-sap — APPROVED** [review]

File: api/cloudcontroller/ccv3/buildpack.go [hunk 1]
```diff
@@ -173,98 +60,29 @@ func (client Client) UpdateBuildpack(buildpack Buildpack) (Buildpack, Warnings,
...
-	}
-
-	var responseBuildpack Buildpack
-	response := cloudcontroller.Response{
-		DecodeJSONResponseInto: &responseBuildpack,
-	}
-	err = client.connection.Make(request, &response)
 
-	return responseBuildpack, response.Warnings, err
+	return responseBody, warnings, err
```
Comment (sleungcy-sap) [inline api/cloudcontroller/ccv3/buildpack.go:62]: actually wouldn't responseBody be undefined here?

**Review by sleungcy-sap — CHANGES_REQUESTED** [review]
seeing one issue

File: api/cloudcontroller/ccv3/buildpack.go (hunk 1 above)
Comment (gmllt) [reply api/cloudcontroller/ccv3/buildpack.go:62]: My bad, thanks for noticing it.
It should be ok now

**Comment by gmllt** [comment]
I've also removed unnecessary definitions for ccv3.APILink, ccv3.APILinks and ccv3.Metadata.
I've updated ccv3 tests accordingly to modifications in ccv3.client.
I don't if updating the v7action ccclient was necessary, so i isolated it in a single commit (in case it's not).

**Comment by sleungcy-sap** [comment]
@loafoe can you help have a look?

**Comment by gmllt** [comment]
@loafoe can you take a look at this PR ? 
We need those changes so we can upgrade the cloudfoundry terraform provider we use to deploy.
Currently, we are locked to [v0.15.4](https://github.com/cloudfoundry-community/terraform-provider-cloudfoundry/releases/tag/v0.15.4).

**Comment by loafoe** [comment]
@gmllt if @sleungcy and @Thanhphan1147 are OK I'm fine merging this one.

**Review by sleungcy — APPROVED** [review]
lgtm, ran a subset of the provder's tests locally with no issues.

**Pull request merged**